scripts/active_sampler_learning_analysis.py

```diff
@@ -0,0 +1,144 @@
+"""Script to analyze samplers learned through the active sampler learning
+approach."""
+
+import os
+from typing import Any, List, Optional, Tuple
+
+import dill as pkl
+import imageio
+import matplotlib.pyplot as plt
+import numpy as np
+from matplotlib import colormaps
+from matplotlib.colors import Normalize
+
+from predicators import utils
+from predicators.envs import BaseEnv, create_new_env
+from predicators.envs.cover import BumpyCoverEnv
+from predicators.settings import CFG
+from predicators.structs import EnvironmentTask, Object, State, Video
+
+
+def _main() -> None:
+    """Loads the saved samplers."""
+    # Parse & validate args
+    args = utils.parse_args()
+    utils.update_config(args)
+    env = create_new_env(CFG.env, do_cache=True)
+    # Set up the test cases.
+    test_cases = _create_test_cases(env)
+    # Set up videos.
+    video_frames = []
+    # Evaluate samplers before offline learning.
+    imgs = _run_one_cycle_analysis(None, test_cases, env)
+    video_frames.append(imgs)
+    # Evaluate samplers for each learning cycle.
+    online_learning_cycle = 1
+    while True:
+        try:
+            imgs = _run_one_cycle_analysis(online_learning_cycle, test_cases,
+                                           env)
+            video_frames.append(imgs)
+        except FileNotFoundError:
+            break
+        online_learning_cycle += 1
+    # Save the videos.
+    for i, video in enumerate(np.swapaxes(video_frames, 0, 1)):
+        video_outfile = f"active_sampler_learning_case_{i}.mp4"
+        utils.save_video(video_outfile, video)
+        # Save the frames individually too.
+        for t, img in enumerate(video):
+            img_outfile = f"videos/active_sampler_learning_case_{i}_{t}.png"
+            imageio.imsave(img_outfile, img)
+
+
+def _create_test_cases(env: BaseEnv) -> List[Tuple[State, List[Object]]]:
+    assert isinstance(env, BumpyCoverEnv)
```

Comment:
Could we instead do an `if isinstance(env, BumpyCoverEnv): ... else: raise NotImplementedError`? Just to make it nicer to implement more envs in this script in the future.